src/braket/default_simulator/openqasm/program_context.py

```diff
@@ -113,29 +115,73 @@ def _(self, identifier: IndexedIdentifier) -> tuple[int]:
         corresponding to the elements referenced by the indexed identifier.
         """
         name = identifier.name.name
+        indices = self.get_qubit_indices(identifier)
+        primary_index = indices[0]
+
+        if isinstance(primary_index, (IntegerLiteral, SymbolLiteral)):
+            target = (self[name][0] + primary_index.value,)
+        elif isinstance(primary_index, RangeDefinition):
+            target = tuple(np.array(self[name])[convert_range_def_to_slice(primary_index)])
+        # Discrete set
+        else:
+            index_list = convert_discrete_set_to_list(primary_index)
+            target = tuple([self[name][0] + index for index in index_list])
+
+        if len(indices) == 2:
+            # used for gate calls on registers, index will be IntegerLiteral
+            secondary_index = indices[1].value
+            target = (target[secondary_index],)
+
+        # validate indices manually, since we use addition instead of indexing to
+        # accommodate symbolic indices
+        for q in target:
+            if isinstance(q, int) and (relative_index := q - self[name][0]) >= len(self[name]):
+                raise IndexError(
+                    f"Index {relative_index} out of bounds for qubit '{name}' "
+                    f"with size {len(self[name])}"
+                )
+        return target
+
+    @staticmethod
+    def get_qubit_indices(
+        identifier: IndexedIdentifier,
+    ) -> list[IntegerLiteral | RangeDefinition | DiscreteSet]:
         primary_index = identifier.indices[0]
         if isinstance(primary_index, list):
             if len(primary_index) != 1:
                 raise IndexError("Cannot index multiple dimensions for qubits.")
             primary_index = primary_index[0]
-        if isinstance(primary_index, IntegerLiteral):
-            target = (self[name][primary_index.value],)
-        elif isinstance(primary_index, RangeDefinition):
-            target = tuple(np.array(self[name])[convert_range_def_to_slice(primary_index)])
-        # Discrete set
-        else:
-            target = tuple(np.array(self[name])[convert_discrete_set_to_list(primary_index)])
 
         if len(identifier.indices) == 1:
-            return target
+            return [primary_index]
         elif len(identifier.indices) == 2:
```

Comment:
Do we need to check, like `primary_index`,  if the `secondary_index` is a List and validate its length?

Reply:
Or it's not revenant because it's always an integerliteral?

Reply:
Technically there's a possibility for something like `qs[0:10][0:5]`, but that is not currently supported. I believe that would throw an error somewhere else, but I could verify. For now, I don't think this is much of a priority. The main (only) place where we see double indices for qubits so far is with `h qs[0:3]` would get mapped to four instructions: `h qs[0:3][0]`, ..., `h qs[0:3][3]`

We could add that validation though anyways

Reply:
I see, I agree that `qs[0:10][0:5]` is an edge use case. We don't need to prioritize the validation I purposed. 

If you were adding the validation, I suggest adding for all elements in `identifier.indices`. I can imagine if `qs[0:10][0:5]` is allowed, users would do `qs[0:10][0:5][1:3]` and so on.